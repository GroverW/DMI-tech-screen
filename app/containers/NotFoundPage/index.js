/**
 * NotFoundPage
 *
 * This is the page we show when the user visits a url that doesn't have a route
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';
import { createSelector } from 'reselect';
import { FormattedMessage } from 'react-intl';
import BorderLink from '../../components/BorderLink';

import { makeGetTheme } from '../Themes/selectors';
import messages from './messages';

function NotFound(props) {
  return (
    <>
      <h1>
        <FormattedMessage {...messages.header} />
      </h1>
      <p>
        <BorderLink to="/" theme={props.theme}>
          <FormattedMessage {...messages.buttonText} />
        </BorderLink>
      </p>
    </>
  );
}

NotFound.propTypes = {
  theme: PropTypes.object,
};

const mapStateToProps = createSelector(
  makeGetTheme(),
  theme => ({
    theme: theme.colors,
  }),
);

export default connect(mapStateToProps)(NotFound);
