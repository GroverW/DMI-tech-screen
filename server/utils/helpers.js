const { RESULTS_PER_PAGE } = require('../config');

const getPageStart = page =>
  page && Number.isInteger(+page)
    ? Math.max(page - 1, 0) * RESULTS_PER_PAGE
    : 0;

module.exports = {
  getPageStart,
};
