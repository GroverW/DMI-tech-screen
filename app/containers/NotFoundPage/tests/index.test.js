import React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import configureStore from '../../../configureStore';
import { renderWithRouter } from '../../../tests/helpers/testHelper';

import NotFoundPage from '../index';

describe('<NotFoundPage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = renderWithRouter(
      <Provider store={store}>
        <IntlProvider locale="en">
          <NotFoundPage />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
