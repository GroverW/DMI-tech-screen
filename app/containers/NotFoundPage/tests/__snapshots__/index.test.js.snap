// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> should render and match the snapshot 1`] = `
<h1>
  <span>
    Sorry, that page doesn't exist!
  </span>
</h1>
`;
